R/utils.R

```diff
@@ -1612,6 +1612,8 @@ URLencode <- function(value, reserved = FALSE) {
 # so that strings like "2016-08-9" are expanded to "2016-08-09"
 dateYMD <- function(date = NULL, argName = "value") {
   if (!length(date)) return(NULL)
+  # as.Date("") yields NA, but we want it to produce an error (so we get the warning)
+  date[!nzchar(date)] <- " "
   tryCatch(date <- format(as.Date(date), "%Y-%m-%d"),
```

Comment:
Instead of special-casing `""`, how about something like this:

```suggestion
  tryCatch({
      date <- format(as.Date(date), "%Y-%m-%d")
      if (is.na(date)) stop()
    },
```

That way, any value that would turn into `NA` will result in the warning. (For example, if input is `NA`, then this would catch it and give a warning.)

Note: I haven't tested this code in action.

Reply:
Another note: we need to make sure this works correctly for vector inputs (see #3061). My initial suggestion above didn't take that into account, but I've since fixed it.